### PR TITLE
Upgrading ember-cli

### PR DESCRIPTION
**Pull request by arthirm**
This PR bumps ember-cli version to 3.5 and fixes the test needed for this change.

@rwjblue

### CODE REVIEW

**Comment by arthirm** [comment]
@Turbo87 Yes am aware of that. Most of it comes from updating ember-cli. Do you suggest doing these changes differently?

**Comment by Turbo87** [comment]
I personally always feel very uncomfortable merging PRs with many independent changes as it makes finding bugs infinitely harder when using something like git bisect. 

IMHO splitting up the PR would be best, possibly in the steps that you mentioned in the PR description. I'd start with the CI change, then drop Node 4 and then continue from there.

**Comment by rwjblue** [comment]
The first one landed in @turbo87’s recent PR merging. Next up is bower support...

**Comment by arthirm** [comment]
Sounds Good! Created a separate PR to remove bower and to run node-tests in CI. Had to club these 2 in a single PR since running node-tests in CI was dependent on removing bower. Also running node-tests while removing bower will make sure that things work fine. 

I will use this PR to upgrade ember-cli alone.

File: .travis.yml [hunk 1]
```diff
@@ -25,17 +38,22 @@ matrix:
...
-  - ember try:one $EMBER_TRY_SCENARIO test --skip-cleanup
-  - yarn test:node
+  - node_modules/.bin/ember try:one $EMBER_TRY_SCENARIO --skip-cleanup
+  - yarn run test:node
+
+language: node_js
+node_js:
+  - "6"
+  - "8"
+  - "10"
```
Comment (Turbo87) [inline .travis.yml]: duplicate `node_js` key?

File: LICENSE.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 The MIT License (MIT)
 
-Copyright (c) 2015
+Copyright (c) 2018
```
Comment (Turbo87) [inline LICENSE.md]: should probably be `2015-2018`?

File: package.json [hunk 3]
```diff
@@ -2,64 +2,70 @@
...
-  "directories": {
-    "doc": "doc",
-    "test": "tests"
-  },
   "scripts": {
     "build": "ember build",
     "changelog": "lerna-changelog",
-    "lint:js": "eslint ./*.js addon addon-test-support app config lib server test-support tests",
-    "start": "ember server",
-    "test": "yarn run test:ember && yarn run test:node",
```
Comment (Turbo87) [inline package.json]: was this change intended?

File: package.json [hunk 4]
```diff
@@ -2,64 +2,70 @@
...
     "ember-addon",
     "ember-cli"
   ],
+  "repository": "https://github.com/ember-cli/ember-cli-htmlbars-inline-precompile",
+  "license": "MIT",
+  "author": "Clemens Müller <cmueller.418@gmail.com>",
+  "directories": {
+    "doc": "doc",
+    "test": "tests"
+  },
```
Comment (Turbo87) [inline package.json]: @kellyselden are you not using `sort-package-json` in `ember-cli-update`?

File: LICENSE.md (hunk 2 above)
Comment (arthirm) [reply LICENSE.md]: The upgrade changed it to 2018. Should it be 2015-2018?

File: package.json (hunk 3 above)
Comment (arthirm) [reply package.json]: It was part of the upgrade.
